src/libcore/tests/iter.rs

```diff
@@ -1428,9 +1470,10 @@ fn test_range_inclusive_nth() {
     assert_eq!(r.nth(2), Some(15));
     assert_eq!(r, 16..=20);
     assert_eq!(r.is_empty(), false);
+    assert_eq!(ExactSizeIterator::is_empty(&r), false);
     assert_eq!(r.nth(10), None);
     assert_eq!(r.is_empty(), true);
-    assert_eq!(r, 1..=0);  // We may not want to document/promise this detail
+    assert_eq!(ExactSizeIterator::is_empty(&r), true);
```

Comment:
Changed RangeInclusive exhaustion tests to check `is_empty`, updated the docs to unspecify the post-iteration value of a RangeInclusive, and added some more Range tests using its `is_empty`.